tuf/repository_lib.py

```diff
@@ -121,11 +121,18 @@ def _generate_and_write_metadata(rolename, metadata_filename, write_partial,
   # Retrieve the roleinfo of 'rolename' to extract the needed metadata
   # attributes, such as version number, expiration, etc.
   roleinfo = tuf.roledb.get_roleinfo(rolename) 
-
+  previous_keyids = roleinfo.get('previous_keyids', [])
+  previous_threshold = roleinfo.get('previous_threshold', 1)
+  
   # Generate the appropriate role metadata for 'rolename'. 
-  if rolename == 'root':
+  if rolename == 'root': 
+    tuf.roledb.update_roleinfo(rolename, roleinfo)
```

Comment:
Not sure why `update_roleinfo()` is needed here.  `roleinfo` isn't updated in any of previous lines.

```
  roleinfo = tuf.roledb.get_roleinfo(rolename) 
  previous_keyids = roleinfo.get('previous_keyids', [])
  previous_threshold = roleinfo.get('previous_threshold', 1)
  
  # Generate the appropriate role metadata for 'rolename'. 
  if rolename == 'root': 
    tuf.roledb.update_roleinfo(rolename, roleinfo)
```

`update_roleinfo()` isn't called like this any of the if-clauses for the other top-level rolenames.